website/blog/2024-04-11-thalium.md

```diff
@@ -0,0 +1,619 @@
+
+This means we have two settings for Java:
+
+- `metals.javaVersion` - version of Java to be used for running Metal language
+  server, used within Visual Studio Code. Users can choose from 11, 17 and 21.
+  Java 8 is deprecated and will be dropped with the next release.
+- `metals.javaHome` - Java used for your project, Metals will try to add all the
+  appropriate flags so that language features are consistent with the version of
+  Java you are using. Will always support JDK 8 and higher.
+
+## Automatic imports setting
+
+Thanks to [keirlawson](http://github.com/keirlawson) it's now possible to make
+Metals automatically import any workspace. This takes a shape of a new setting
+`metals.autoImportBuild` which has three options:
+
+- `off` - users will always be asked whether they want to import their workspace
+  either initially or after changes.
+- `initial` - when users first open a new workspace Metals will automatically
+  import the workspace, but will still ask about reimporting after changes.
+- `all` - Metals will always automatically import and reimport your workspace.
+  This might make more sense when using other build tools than Bloop as it might
+  take a while to run the import.
+
+## Preferred build server setting
+
+Thanks to [kasiaMarek](https://github.com/kasiaMarek) it's now possible to
+default to the BSP server instead of Bloop for all your workspaces using
+`metals.defaultBspToBuildTool` setting. This will work with both Mill and sbt,
+which support both.
+
+## Inlay hints support
+
+After a huge amount of work by [jkciesluk](https://github.com/jkciesluk) Metals
+will not use the standard LSP way of showing decorations via Inlay Hints instead
+of a custom solution. Thanks to this, the decorations behave much more naturally
+like usual code, you can navigate to definition and get proper hovers.
+
+![inlay-hints](https://i.imgur.com/G6H2aPr.gif)
+
+Additionally, together with inlay hints we reworked all the decoration settings
+to be included in a single object and two new ones were added:
+
+- `metals.inlayHints.typeParameters.enable` - when this option is enabled, each
+  place when a type parameter is applied has it displayed either as additional
+  decorations if they are supported by the editor or shown in the hover.
+- `metals.inlayHints.hintsInPatternMatch.enable` - when this option is enabled,
+  each place when a type is inferred in a pattern match has it displayed either
+  as additional decorations if they are supported by the editor or shown in the
+  hover.
+
+There is a small issue within VS Code, which might be fixed in the future, and
+it makes it impossible to hover on multiple parts of a particular type
+consecutively. Users have to move the cursor away from the type and again onto
+the part of the type they want to learn more about.
+
+Inlay hints will also not work 100% for any older deprecated Scala 2 versions
+and Scala 3 versions 3.4.0, 3.4.1, 3.3.0, 3.3.1, 3.3.2, 3.3.3. For those
```

Comment:
So they work 100% on 3.4.0 and later?

Reply:
As I checked they work fully for 3.4.2 and later, probably also next LTS. This is because we needed to do a change inside the compiler